src/core/transpile.jl

```diff
@@ -2064,3 +2101,68 @@ function transpile(
 
     return circuit
 end
+
+struct DecomposeControlledGatesTranspiler <: Transpiler end
+
+function transpile(
+    ::DecomposeControlledGatesTranspiler,
+    circuit::QuantumCircuit,
+)::QuantumCircuit
+    qubit_count = get_num_qubits(circuit)
+    output = QuantumCircuit(qubit_count = qubit_count, name = get_name(circuit))
+
+    for instr in get_circuit_instructions(circuit)
+        if instr isa Readout
+            push!(output, instr)
+            continue
+        end
+
+        gate = get_gate_symbol(instr)
+        if gate isa Controlled
+            kernel = get_kernel(gate)
+
+            connected_qubits = get_connected_qubits(instr)
+            if length(connected_qubits) != 2
+                throw(
+                    ArgumentError(
+                        "DecomposeControlledGatesTranspiler is only implemented for single-target single-control Controlled Gates",
+                    ),
+                )
+            end
+
+            if typeof(kernel) == SigmaZ
```

Comment:
Perhaps we should just push a `control_z` gate here and let a later stage transpile that.

Reply:
updated